src/main/java/Aaron/Task/Task.java

```diff
@@ -0,0 +1,62 @@
+package Aaron.Task;
+import Aaron.Exception.DoubleMarkException;
+import Aaron.Exception.TaskNoNameException;
+
+/**
+ * class that represents a task instantiated in the tasklist of aaronbot
+ */
+public abstract class Task {
+    private boolean isDone;
+    private String taskString;
+
+    public Task(String taskString) throws TaskNoNameException {
+        this.isDone = false;
+        if (taskString.trim().isEmpty()) {
+            throw new TaskNoNameException("STUDENT!! You need a task description :)");
+        } else {
+            this.taskString = taskString;
+        }
+    }
+
+    public Task(String taskString, boolean isDone) {
+        this.isDone = isDone;
+        this.taskString = taskString;
+    }
+
+    public void markDone() throws DoubleMarkException{
+        if (isDone) {
+            throw new DoubleMarkException("Task is already marked \n " + this.toString());
+        } else { 
+            isDone = true;
+        }
+    }
+
+    public boolean unmarkDone() throws DoubleMarkException{
+        if (!isDone) {
+            throw new DoubleMarkException("Task is already unmarked \n " + this.toString());
+        } else {
+            isDone = false;
+            return true;
+        }
+    }
+
```

Comment:
I like that you have custom exceptions which are named appropriately following the convention of ending with 'Exception'